enterprise/server/remote_execution/containers/podman/podman.go

```diff
@@ -93,17 +102,37 @@ func (c *podmanCommandContainer) Run(ctx context.Context, command *repb.Command,
 }
 
 func (c *podmanCommandContainer) Create(ctx context.Context, workDir string) error {
-	c.workDir = workDir
-	return nil
+	containerName, err := generateContainerName()
+	if err != nil {
+		return status.UnavailableErrorf("failed to generate podman container name: %s", err)
+	}
+	c.name = containerName
+
+	podmanRunArgs := c.getPodmanRunArgs(workDir, false /*=autoRemove*/)
+	podmanRunArgs = append(podmanRunArgs, c.image)
+	podmanRunArgs = append(podmanRunArgs, "sleep", "infinity")
+	createResult := runPodman(ctx, "create", nil, nil, podmanRunArgs...)
+	if err = createResult.Error; err != nil {
+		return status.UnavailableErrorf("failed to create container: %s", err)
+	}
+
+	startResult := runPodman(ctx, "start", nil, nil, c.name)
+	return startResult.Error
 }
 
 func (c *podmanCommandContainer) Exec(ctx context.Context, cmd *repb.Command, stdin io.Reader, stdout io.Writer) *interfaces.CommandResult {
-	return runPodman(ctx, "run" /*workDir=*/, "", stdin, stdout, cmd.Arguments...)
+	podmanRunArgs := []string{}
+	for _, envVar := range cmd.GetEnvironmentVariables() {
+		podmanRunArgs = append(podmanRunArgs, "--env", fmt.Sprintf("%s=%s", envVar.GetName(), envVar.GetValue()))
+	}
+	podmanRunArgs = append(podmanRunArgs, c.name)
+	podmanRunArgs = append(podmanRunArgs, cmd.Arguments...)
+	return runPodman(ctx, "exec", stdin, stdout, podmanRunArgs...)
 }
 
 func (c *podmanCommandContainer) IsImageCached(ctx context.Context) (bool, error) {
 	// Try to avoid the `pull` command which results in a network roundtrip.
-	listResult := runPodman(ctx, "images", "", nil, nil, "--filter=reference="+c.image, "--format={{.ID}}")
+	listResult := runPodman(ctx, "images", nil, nil, "--filter=reference="+c.image, "--format={{.ID}}")
```

Comment:
nit: I think the `--filter` here does a string-contains check, so if there is a cached image tagged `v1.1` and we check whether `v1` is cached, it will incorrectly return true -- might want to split the output lines into columns and then return whether at least one line matches an `==` check on the image ref column.

Reply:
Done. I actually use podman image inspect. inspect command can be used to inspect containers, volumes, network, and pods.